var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py

```diff
@@ -129,6 +129,7 @@ def setup_run_environment(self, env):
 
     def setup_dependent_build_environment(self, env, dependent_spec):
         env.set("MKLROOT", self.component_prefix)
+        # env.append_path("PKG_CONFIG_PATH", self.component_prefix.lib.pkgconfig)
```

Comment:
Did something happen during a rebase? This PR seems just to add a commented out line?

Reply:
I believe this was designed to test whether or not the feature added in this PR was the reason that CI fails 99% of the time.

Reply:
@rscohn2 can you uncomment this? It seems like CI is working again.

Reply:
It passed when I commented out the line because it did not run any of the tests. The logic that decides when to run tests must be very smart... or broken.

Reply:
Does it compute a hash that is not affected by comments?

Reply:
Yes, it is very smart. I believe @tgamblin wrote all the logic that goes into computing the hash, he might be able to point to the logic. At a very minimum, we convert to an AST (so comments are stripped), and we manually remove things like docstrings that won't affect the build process.

Reply:
You can see the source that affects the hash with:

```console
spack pkg source --canonical intel-oneapi-mkl
```

and the hash with 

```console
spack pkg hash intel-oneapi-mkl
```

Reply:
The tests might be the most intuitive way to see what's happening: https://github.com/spack/spack/blob/develop/lib/spack/spack/test/util/package_hash.py#L154-L456